server/src/main/java/org/spine3/server/integration/IntegrationEventBus.java

```diff
@@ -20,39 +20,136 @@
 
 package org.spine3.server.integration;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
 import com.google.protobuf.Message;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spine3.base.Event;
+import org.spine3.base.Response;
 import org.spine3.server.BoundedContext;
 import org.spine3.server.event.EventBus;
+import org.spine3.server.event.Subscribe;
+import org.spine3.server.type.EventClass;
+
+import java.util.Collection;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spine3.protobuf.Messages.fromAny;
+import static org.spine3.server.integration.IntegrationEventSubscriberGrpc.IntegrationEventSubscriber;
+import static org.spine3.validate.Validate.checkNotEmptyOrBlank;
 
 /**
+ * Allows to register integration event subscribers and deliver events to them.
+ *
+ * <p>An integration event is sent between loosely coupled parts of a system.
+ * Typically such parts would be implemented as {@link BoundedContext}s.
  *
+ * @see Subscribe
+ * @see EventBus
  */
 public class IntegrationEventBus {
 
-    private final EventBus bus;
+    private final Multimap<EventClass, IntegrationEventSubscriber> subscribersMap = HashMultimap.create();
 
-    public IntegrationEventBus() {
-        //TODO:2016-04-27:alexander.yevsyukov: Implement.
-        bus = null;
+    private StreamObserver<Response> responseObserver = new DefaultResponseObserver();
+
+    /**
+     * Returns an event bus instance.
+     */
+    public static IntegrationEventBus getInstance() {
+        return instance();
     }
 
-    public void post(IntegrationEvent event) {
-        //TODO:2016-04-27:alexander.yevsyukov: Implement
+    /**
+     * Sets a response observer used in {@link IntegrationEventSubscriber#notify(Event, StreamObserver)}.
+     */
+    public void setResponseObserver(StreamObserver<Response> responseObserver) {
```

Comment:
The response observer is passed with the call. We don't set it. See gRPC examples.